main.js

```diff
@@ -0,0 +1,77 @@
+let books = [];
+
+const book = document.getElementById('book');
+
+function Book(title, author) {
+  this.title = title;
+  this.author = author;
+}
+
+function addBook() {
+  for (let i = 0; i < books.length; i += 1) {
+    const bookData = `
+        <div id="book${i}">
+            <h2>${books[i].title}</h2>
+            <p>${books[i].author}</p>
+            <button id="${i}" class="remove">x</button>
+            <hr>
+        </div>
+        `;
+    book.innerHTML += bookData;
+  }
+}
+
+addBook();
+
+const forma = document.querySelector('form');
+const suma = 1;
+
+forma.addEventListener('submit', (e) => {
+  e.preventDefault();
+  let id = 0;
+
+  if (books.length === 0) {
+    id = 1;
+  } else {
+    id = [(books.length - 1) + suma];
+  }
+  const t = document.getElementById('title').value;
+  const a = document.getElementById('author').value;
+  const y = new Book(t, a);
+  books.push(y);
+  const bookData = `
+    <div id="${id}">
+        <h2>${y.title}</h2>
+        <p>${y.author}</p>
+        <button id="${y}" class="remove">x</button>
+        <hr>
+    </div>
+    `;
+  book.innerHTML += bookData;
+  const titleValue = document.getElementById('title');
+  const authorValue = document.getElementById('author');
+  titleValue.value = '';
+  authorValue.value = '';
+  localStorage.setItem('books', JSON.stringify(books));
+});
+
+book.addEventListener('click', (e) => {
+  if (e.target.textContent === 'x') {
+    const booksId = e.target.parentElement;
+    const { id } = e.target;
+    books.splice(id, 1);
+    booksId.remove();
+  }
+  localStorage.setItem('books', JSON.stringify(books));
+});
```

Comment:
- [ ] Kindly do the same thing with the remove book function as i suggested you to do with the add a single book function above ⬆️ to make your code reusable and clean 😊

Reply:
Hello @HAMMAS-SALEEM thank you very much for your suggestion, I was implementing what you told me and added AddSingleBook as you suggested.
![AddSingleBooks](https://user-images.githubusercontent.com/90111138/204661982-de478945-da6f-4262-952f-5390db5c8a40.png)

But I was wondering if this is what you meant with the section where I remove a book.
![Remove Books](https://user-images.githubusercontent.com/90111138/204662149-f0004823-d5e2-4ac5-bb22-224c7c98d96f.png)
Anyway, I really appreciated your support especially since a lot of this is new to me and my partner.

Reply:
Hey @HunterX-7, @emmanuelcortes2 :wave:

You guys can make this remove function as shown below :+1: :arrow_down:
```javascript

const removeSingleBook = (e) => {
  if (e.target.textContent === 'x') {
    const booksId = e.target.parentElement;
    const { id } = e.target;
    books.splice(id, 1);
    booksId.remove();
  }
  localStorage.setItem('books', JSON.stringify(books));
}

book.addEventListener('click', (e) => {
  removeSingleBook(e)
});

```